pyzo/codeeditor/extensions/behaviour.py

```diff
@@ -441,3 +442,90 @@ def _paste(self, keepSelection):
             cursor.setKeepPositionOnInsert(False)
             self.setTextCursor(cursor)
 
+
+class AutoCloseQuotesAndBrackets(object):
+    """
+    Automatic insertion of quotes, parenthesis, braces and brackets
+
+    """
+    
+    def keyPressEvent(self, event):
+        
+        quotesKeys = [Qt.Key_Apostrophe, Qt.Key_QuoteDbl]
+        openBrackets = [Qt.Key_BraceLeft,Qt.Key_BracketLeft, Qt.Key_ParenLeft]
+        closeBrackets = [Qt.Key_BraceRight, Qt.Key_BracketRight, Qt.Key_ParenRight]
+
+        cursor = self.textCursor()
+        
+        # opening brackets
+        if event.key() in openBrackets and pyzo.config.settings.autoClose_Brackets:
+            idx = openBrackets.index(event.key())
+            insert_txt = "{}{}".format(chr(openBrackets[idx]), chr(closeBrackets[idx]))
+            cursor.insertText(insert_txt)
+            self._moveCursorLeft(1)
+        
+        # closing brackets
+        elif event.key() in closeBrackets and pyzo.config.settings.autoClose_Brackets:
+            idx = closeBrackets.index(event.key())
+            next_character = self._getNextCharacter(cursor)
+            if next_character:
+                # skip
+                if ord(next_character) in closeBrackets:
+                    self._moveCursorRight(1)
+                # close
+                else:
+                    insert_txt = chr(event.key())
+                    cursor.insertText(insert_txt)
+            else:
+                super().keyPressEvent(event)
+        
+        # quotes
+        elif event.key() in quotesKeys and pyzo.config.settings.autoClose_Quotes:
+            idx = quotesKeys.index(event.key())
+            next_character = self._getNextCharacter(cursor)
+            if next_character:
+                # skip
+                if ord(next_character) == event.key():
+                    self._moveCursorRight(1)
+                # close
+                else:
+                    insert_txt = "{}{}".format(chr(quotesKeys[idx]), chr(quotesKeys[idx]))
+                    cursor.insertText(insert_txt)
+                    self._moveCursorLeft(1)
+            else:
+                insert_txt = "{}{}".format(chr(quotesKeys[idx]), chr(quotesKeys[idx]))
+                cursor.insertText(insert_txt)
+                self._moveCursorLeft(1)
+                # triple-quoted
+                cursor2 = self.textCursor()
+                textBeforeCursor = ustr(cursor2.block().text())[:cursor2.positionInBlock()]
+                if textBeforeCursor.endswith(chr(event.key())*3):
+                    cursor2.insertText(insert_txt)
+                    self._moveCursorLeft(2)
+        else:
+            super().keyPressEvent(event)
+
+    def _getNextCharacter(self, cursor):
```

Comment:
It's probably good to use double underscore for these method names to avoid potential name clashes with other editor plugins.